server/src/main/java/de/uftos/repositories/ucdl/parser/DefinitionParser.java

```diff
@@ -0,0 +1,691 @@
+                  || (filter.getToken() == UcdlToken.RESOURCE_SET
+                  && ((SetDto) filter).type() != setType)) {
+                throw new ParseException("Sets can only be filtered using sets of the same type!");
+              }
+            }
+
+            modifiers.add(new OperatorDto(UcdlToken.FILTER, filters));
+            modifierList = modifierList.jjtGetChild(2);
+            parameters.replace("this", thisType);
+          }
+        }
+
+        return new SetDto(UcdlToken.RESOURCE_SET, setType, setName, modifiers);
+      }
+      case "NUMBERSET" -> {
+        List<Integer> values = new ArrayList<>();
+
+        values.add(Integer.parseInt(((SimpleNode) root).jjtGetFirstToken().next.image));
+
+        SimpleNode numberList = (SimpleNode) root.jjtGetChild(0);
+        while (numberList.jjtGetNumChildren() > 0) {
+          values.add(Integer.parseInt(numberList.jjtGetFirstToken().next.image));
+          numberList = (SimpleNode) numberList.jjtGetChild(0);
+        }
+        return new ValueDto<>(UcdlToken.NUMBER_SET, values.toArray(new Integer[0]));
+      }
+      case "VALUEREFERENCE" -> {
+        return new ValueDto<>(UcdlToken.VALUE_REFERENCE,
+            ((SimpleNode) root).jjtGetFirstToken().image);
+      }
+      case "FILTER" -> {
+        switch (root.jjtGetChild(0).toString()) {
+          case "BOOLVALUE":
+          case "FORALL":
+          case "EXISTS":
+          case "ISEMPTY":
+            return buildAst(root.jjtGetChild(0), parameters);
+          case "NUMBERSET":
+            AbstractSyntaxTreeDto numberSet = buildAst(root.jjtGetChild(0), parameters);
+
+            ResourceType thisType = parameters.get("this");
+            parameters.replace("this", ResourceType.NUMBER);
+
+            List<AbstractSyntaxTreeDto> modifiers = new ArrayList<>();
+
+            Node modifierList = root.jjtGetChild(1);
+            while (modifierList.jjtGetNumChildren() > 0) {
+              if (modifierList.jjtGetNumChildren() == 2) { //-> Flatmap
+                throw new ParseException("Numbers don't have any attributes!");
+              } else if (modifierList.jjtGetNumChildren() == 3) {
+                List<AbstractSyntaxTreeDto> filters = new ArrayList<>();
+                filters.add(buildAst(modifierList.jjtGetChild(0), parameters));
+
+                Node filterList = modifierList.jjtGetChild(1);
+                while (filterList.jjtGetNumChildren() > 0) {
+                  filters.add(buildAst(filterList.jjtGetChild(0), parameters));
+                  filterList = filterList.jjtGetChild(1);
+                }
+                for (AbstractSyntaxTreeDto filter : filters) {
+                  if (filter.getToken() == UcdlToken.RESOURCE_SET
+                      && ((SetDto) filter).type() != ResourceType.NUMBER) {
+                    throw new ParseException(
+                        "ets can only be filtered using sets of the same type!");
```

Comment:
`ets`?

Reply:
I fixed this typo once before...
I guess it was a different branch...

Reply:
In fact: it was the solver branch...

Reply:
Got it